bridge/client/app/_views/ktb-environment-view/ktb-environment-view.component.ts

```diff
@@ -1,29 +1,86 @@
-import { filter, map, switchMap } from 'rxjs/operators';
-import { Observable } from 'rxjs';
-import { ActivatedRoute } from '@angular/router';
+import { Component, HostBinding, Inject, OnDestroy } from '@angular/core';
+import { distinctUntilChanged, filter, map, switchMap, takeUntil } from 'rxjs/operators';
+import { combineLatest, Observable, Subject } from 'rxjs';
+import { ActivatedRoute, Router } from '@angular/router';
 import { Project } from '../../_models/project';
 import { DataService } from '../../_services/data.service';
+import { AppUtils, POLLING_INTERVAL_MILLIS } from '../../_utils/app.utils';
+import { ServiceFilterType } from './ktb-stage-details/ktb-stage-details.component';
+import { Stage } from '../../_models/stage';
+import { Location } from '@angular/common';
+
+export interface ISelectedStageInfo {
+  stage: Stage;
+  filterType: ServiceFilterType;
+}
 
 @Component({
   selector: 'ktb-environment-view',
   templateUrl: './ktb-environment-view.component.html',
   styleUrls: ['./ktb-environment-view.component.scss'],
   preserveWhitespaces: false,
 })
-export class KtbEnvironmentViewComponent {
+export class KtbEnvironmentViewComponent implements OnDestroy {
   @HostBinding('class') cls = 'ktb-environment-view';
-  public project$: Observable<Project | undefined>;
+  public project$: Observable<Project>;
+  private readonly unsubscribe$ = new Subject<void>();
+  public selectedStageInfo?: ISelectedStageInfo;
+
+  constructor(
+    private dataService: DataService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private location: Location,
+    @Inject(POLLING_INTERVAL_MILLIS) private initialDelayMillis: number
+  ) {
+    const selectedStageName$ = this.route.paramMap.pipe(
+      map((params) => params.get('stageName')),
+      takeUntil(this.unsubscribe$)
+    );
 
-  constructor(private dataService: DataService, private route: ActivatedRoute) {
+    const paramFilterType$ = this.route.queryParamMap.pipe(map((params) => params.get('filterType')));
     const projectName$ = this.route.paramMap.pipe(
       map((params) => params.get('projectName')),
-      filter((projectName): projectName is string => !!projectName)
+      filter((projectName): projectName is string => !!projectName),
+      distinctUntilChanged()
     );
 
     this.project$ = projectName$.pipe(
       switchMap((projectName) => this.dataService.getProject(projectName)),
-      map((project) => (project?.projectDetailsLoaded ? project : undefined))
+      filter((project): project is Project => !!project?.projectDetailsLoaded)
+    );
+
+    combineLatest([selectedStageName$, paramFilterType$, this.project$])
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(([stageName, filterType, project]) => {
+        const stage = project.stages.find((s) => s.stageName === stageName);
+        if (stage) {
+          this.selectedStageInfo = { stage: stage, filterType: (filterType as ServiceFilterType) ?? undefined };
```

Comment:
The `subscribe` would still be here even if we set the `selectedStageInfo$` to an observable because there is another setter for this one. `selectedStageInfo` does not only initially depend on the route params and the project, but also on the child component that emits the changed object. Then `selectedStageInfo$` will be a BehaviorSubject where we call `next(...)` on change of the child component that emits the new value and then we will again need the subscribe in order to set the initial value for the BehaviorSubject.

Reply:
After discussion IRL we decided to keep it in as is and create a follow-up ticket for the router/routing/re-rendering situation.